### PR TITLE
Add Bucket#policy_only? and #policy_only=

### PR DESCRIPTION
**Pull request by quartzmo**
Add support for Bucket Policy Only.

### CODE REVIEW

**Comment by quartzmo** [comment]
Storage passed in Kokoro Linux although several other packages failed:

> [10:49:25] Child job: cloud-devrel/client-libraries/google-cloud-ruby/presubmit/osx/storage finished with status: SUCCEEDED

**Comment by quartzmo** [comment]
Ping @frankyn

File: google-cloud-storage/lib/google/cloud/storage/bucket.rb [hunk 1]
```diff
@@ -757,6 +760,71 @@ def lock_retention_policy!
...
+        #
+        # @example
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #
+        #   bucket = storage.bucket "my-bucket"
+        #
+        #   bucket.policy_only = true
+        #   bucket.policy_only? # true
```
Comment (frankyn) [inline google-cloud-storage/lib/google/cloud/storage/bucket.rb]: IAMConfiguration in the future is expected to have additional fields. Given the current design do you feel this is manageable?

Comment (frankyn) [inline google-cloud-storage/lib/google/cloud/storage/bucket.rb]: Please expose the field: `iamConfiguration.lockedTime`. A user should be able to check when this field can no longer be reverted.

Comment (quartzmo) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: Flattening is well established in the design for configuration objects with a couple of fields (`Bucket#website_main=` and `Bucket#website_404` for example.) Do you anticipate a lot of complexity in IAMConfiguration?

Comment (frankyn) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: Based on prior conversations it should be similar fields that will be exposed here in the future.

Sgtm, I just wanted to double check. Thank you for the feedback.

Comment (quartzmo) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: @frankyn @blowmage I exposed `lockedTime` as `Bucket#policy_only_locked_at`. This name complies with the existing naming convention for DateTime fields, such as `Bucket#retention_effective_at`.

**Comment by quartzmo** [comment]
@frankyn Updated, PTAL!

File: google-cloud-storage/lib/google/cloud/storage/bucket.rb (hunk 1 above)
Comment (blowmage) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: 👍

Comment (frankyn) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: lgtm!

Comment (frankyn) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: One nit: @quartzmo please add `bucket.retention_effective_at?` into the example above to relate the methods.

**Comment by frankyn** [comment]
Let's follow-up on this question in the next weekly meeting. Thank you @quartzmo! Super fast turn around.

File: google-cloud-storage/lib/google/cloud/storage/bucket.rb (hunk 1 above)
Comment (frankyn) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: @quartzmo, could you add `puts bucket.policy_only_locked_at?` into this example?

Comment (quartzmo) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: Updated.

**Comment by frankyn** [comment]
LGTM, thank you @quartzmo. I'll ping this PR when it's ready to merge/release.

**Comment by quartzmo** [comment]
I rebased on master and ran acceptance tests locally with my personal dev service account, passing.

**Comment by frankyn** [comment]
Thanks @quartzmo!

**Comment by blowmage** [comment]
@quartzmo I'm late to this, I know. but is the dependency on google-api-client set to a version that always has `BucketPolicyOnly` defined? Has that been checked?

**Comment by quartzmo** [comment]
@blowmage Thanks, I'll look at that before releasing.

**Comment by frankyn** [comment]
Thanks for pointing that out @blowmage, I'll keep that in mind for next time.

**Comment by quartzmo** [comment]
The dependency spec in [`google-cloud-storage.gemspec`](https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-storage/google-cloud-storage.gemspec#L22) is `"google-api-client", "~> 0.23"`. Version `0.23.0` was released June 04, 2018.

**Comment by blowmage** [comment]
Does that google-api-client release include BucketPolicyOnly?

**Pull request merged**